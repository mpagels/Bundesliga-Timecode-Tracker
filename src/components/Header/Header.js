import React from 'react'
import styled from 'styled-components/macro'
import { ReactComponent as Cross } from '../../assets/cross-white.svg'
import { ReactComponent as Settings } from '../../assets/settings.svg'
import { ReactComponent as Plus } from '../../assets/plus-white.svg'
import TimeCodeBubble from '../TimeCodeBubble/TimeCodeBubble'
import { Link, useHistory } from 'react-router-dom'

export default function Header({
  title,
  type,
  isCloseButton,
  totalLength,
  duration,
  onCircleClick,
}) {
  let history = useHistory()

  return (
    <Wrapper type={type}>
      <Top type={type}>
        <StyledHeader>{title}</StyledHeader>
        <ActionWrapper>
          {isCloseButton ? (
            <Circle onClick={() => history.goBack()}>
              <Cross />
            </Circle>
          ) : (
            <>
              <Circle>
                <Settings />
              </Circle>
              <AddPlus>
                <Link to="/create">
                  <Plus />
                </Link>
              </AddPlus>
            </>
          )}
        </ActionWrapper>
      </Top>

      {type === 'big' && (
        <BubbleWrapper>
          <TimeCodeBubble title="GESAMTLÃ„NGE" timeCode={totalLength} />
          <TimeCodeBubble title="VORGABE" timeCode={duration} />
        </BubbleWrapper>
      )}
    </Wrapper>
  )
}

const Wrapper = styled.div`
  position: fixed;
  top: 0%;
  width: 100%;
  left: 0%;
  height: ${({ type }) => (type === 'small' ? '80px;' : '160px')};
  ${({ type }) => type === 'small' && 'display: flex; align-items: center;'}

  & h1 {
    font-family: 'BaiJamjuree';
    margin-left: 20px;
    color: var(--font-blue);
  }
`
const Top = styled.div`
  width: 100%;
  ${({ type }) => type === 'big' && 'margin-top: 10px;'}
  display: flex;
  align-items: center;
  justify-content: space-between;
`
const StyledHeader = styled.h1`
  font-size: 1.2em;
  margin: 0;
`

const ActionWrapper = styled.div`
  display: flex;
  align-items: center;
`
const BubbleWrapper = styled.div`
  display: flex;
  margin: 10px 10px 0;
`

const Circle = styled.button`
  all: unset;
  cursor: pointer;
  display: flex;
  margin-right: 10px;
  justify-content: center;
  align-items: center;
  width: 45px;
  height: 45px;
  border-radius: 50%;
  background: linear-gradient(180deg, #001496 0%, #0032c8 100%);
`

const AddPlus = styled.button`
  cursor: pointer;
  all: unset;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 50% 0 0 50%;
  height: 60px;
  width: 65px;
  background: linear-gradient(180deg, #001496 0%, #0032c8 100%);
`
