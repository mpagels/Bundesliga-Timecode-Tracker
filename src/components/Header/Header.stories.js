import React from 'react'
import Header from './Header'

export default {
  title: 'capstone/Header',
  component: Header,
}

const Template = (args) => <Header {...args} />

export const SmallWithCloseButton = Template.bind({})
SmallWithCloseButton.args = {
  title: 'NEUE SZENE HINZUFÃœGEN',
  type: 'small',
  isCloseButton: true,
}

export const BigWithButtonsAndTimecodes = Template.bind({})
BigWithButtonsAndTimecodes.args = {
  title: 'TIMECODE TRACKER',
  type: 'big',
  isCloseButton: false,
  totalLength: '00:00:00',
  duration: '00:00:00',
}
