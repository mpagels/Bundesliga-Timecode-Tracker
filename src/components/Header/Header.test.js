import React from 'react'
import Header from './Header'
import { render, cleanup } from '@testing-library/react'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'

afterEach(cleanup)

it('renders correctly a small type header', () => {
  const history = createMemoryHistory()
  const { asFragment, debug } = render(
    <Router history={history}>
      <Header type="small" hasCloseButton={false} />
    </Router>
  )
  expect(asFragment()).toMatchSnapshot()
})

it('renders correctly a big type header', () => {
  const history = createMemoryHistory()
  const { asFragment, debug } = render(
    <Router history={history}>
      <Header
        type="big"
        hasCloseButton={false}
        totalLength="00:00:00:00"
        duration="00:00:00:00"
      />
    </Router>
  )
  expect(asFragment()).toMatchSnapshot()
})

it('renders correctly a header with close button', () => {
  const history = createMemoryHistory()
  const { asFragment, debug } = render(
    <Router history={history}>
      <Header type="small" hasCloseButton={true} />
    </Router>
  )
  expect(asFragment()).toMatchSnapshot()
})
