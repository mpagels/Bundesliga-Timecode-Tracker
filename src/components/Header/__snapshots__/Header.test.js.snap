// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct 1`] = `
<DocumentFragment>
  <div
    class="Header__Wrapper-sc-42wb12-0 kIEjuF"
    type="small"
  >
    <div
      class="Header__Top-sc-42wb12-1 hMdxRz"
      type="small"
    >
      <h1
        class="Header__StyledHeader-sc-42wb12-2 egjaic"
      >
        TIMECODE TRACKER
      </h1>
      <div
        class="Header__ActionWrapper-sc-42wb12-3 bpFiNn"
      >
        <a
          href="/settings"
        >
          <button
            class="Header__Circle-sc-42wb12-5 cbzzXT"
          >
            <svg>
              settings.svg
            </svg>
          </button>
        </a>
        <a
          href="/create"
        >
          <button
            class="Header__AddPlus-sc-42wb12-6 fAopbb"
          >
            <svg>
              plus-white.svg
            </svg>
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders correctly a big type header 1`] = `
<DocumentFragment>
  <div
    class="Header__Wrapper-sc-42wb12-0 bfPzTh"
    type="big"
  >
    <div
      class="Header__Top-sc-42wb12-1 czgLcD"
      type="big"
    >
      <h1
        class="Header__StyledHeader-sc-42wb12-2 egjaic"
      >
        TIMECODE TRACKER
      </h1>
      <div
        class="Header__ActionWrapper-sc-42wb12-3 bpFiNn"
      >
        <a
          href="/settings"
        >
          <button
            class="Header__Circle-sc-42wb12-5 cbzzXT"
          >
            <svg>
              settings.svg
            </svg>
          </button>
        </a>
        <a
          href="/create"
        >
          <button
            class="Header__AddPlus-sc-42wb12-6 fAopbb"
          >
            <svg>
              plus-white.svg
            </svg>
          </button>
        </a>
      </div>
    </div>
    <div
      class="Header__BubbleWrapper-sc-42wb12-4 fpvdLP"
    >
      <div
        class="TimeCodeBubble__BubbleWrapper-sc-1ve62kg-0 ddckpx"
      >
        <p
          class="TimeCodeBubble__ForWhat-sc-1ve62kg-1 bZpzYP"
        >
          GESAMTLÃ„NGE
        </p>
        <p
          class="TimeCodeBubble__TimeCode-sc-1ve62kg-2 hSwuJE"
        >
          00:00:00:00
        </p>
      </div>
      <div
        class="TimeCodeBubble__BubbleWrapper-sc-1ve62kg-0 ddckpx"
      >
        <p
          class="TimeCodeBubble__ForWhat-sc-1ve62kg-1 bZpzYP"
        >
          VORGABE
        </p>
        <p
          class="TimeCodeBubble__TimeCode-sc-1ve62kg-2 hSwuJE"
        >
          00:00:00:00min
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders correctly a header with close button 1`] = `
<DocumentFragment>
  <div
    class="Header__Wrapper-sc-42wb12-0 kIEjuF"
    type="small"
  >
    <div
      class="Header__Top-sc-42wb12-1 hMdxRz"
      type="small"
    >
      <h1
        class="Header__StyledHeader-sc-42wb12-2 egjaic"
      >
        TIMECODE TRACKER
      </h1>
      <div
        class="Header__ActionWrapper-sc-42wb12-3 bpFiNn"
      >
        <button
          class="Header__Circle-sc-42wb12-5 cbzzXT"
        >
          <svg>
            cross-white.svg
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders correctly a small type header 1`] = `
<DocumentFragment>
  <div
    class="Header__Wrapper-sc-42wb12-0 kIEjuF"
    type="small"
  >
    <div
      class="Header__Top-sc-42wb12-1 hMdxRz"
      type="small"
    >
      <h1
        class="Header__StyledHeader-sc-42wb12-2 egjaic"
      >
        TIMECODE TRACKER
      </h1>
      <div
        class="Header__ActionWrapper-sc-42wb12-3 bpFiNn"
      >
        <a
          href="/settings"
        >
          <button
            class="Header__Circle-sc-42wb12-5 cbzzXT"
          >
            <svg>
              settings.svg
            </svg>
          </button>
        </a>
        <a
          href="/create"
        >
          <button
            class="Header__AddPlus-sc-42wb12-6 fAopbb"
          >
            <svg>
              plus-white.svg
            </svg>
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
